system/src/system_control.cpp

```diff
@@ -292,15 +286,51 @@ void SystemControlInterface::setRequestResult(USBRequest* req, USBRequestResult
 
 void SystemControlInterface::processSystemRequest(void* data) {
   USBRequest* req = static_cast<USBRequest*>(data);
+  USBRequestResult result = USB_REQUEST_RESULT_OK;
   switch (req->type) {
   // Handle requests that should be processed by the system modules here
+  case USB_REQUEST_RESET: {
+    System.reset(req->value);
+    break;
+  }
+
+  case USB_REQUEST_DFU_MODE: {
+    System.dfu(false);
+    break;
+  }
+
+  case USB_REQUEST_LISTENING_MODE: {
+    setRequestResult(req, result);
```

Comment:
Is there any problem with setting a result after calling `WiFi.listen()`? I just would like to see error handling unified in this function (if possible), i.e. make all case blocks to either call `setRequestResult()` and return, or assign to `result` variable.